packages/rich-text/src/component/index.js

```diff
@@ -403,17 +404,10 @@ class RichText extends Component {
 	 * @param {WPSyntheticEvent} event Synthetic input event.
 	 */
 	onInput( event ) {
-		// For Input Method Editor (IME), used in Chinese, Japanese, and Korean
-		// (CJK), do not trigger a change if characters are being composed.
-		// Browsers setting `isComposing` to `true` will usually emit a final
-		// `input` event when the characters are composed.
-		if (
-			event &&
-			event.nativeEvent &&
-			event.nativeEvent.isComposing
-		) {
-			// Also don't update any selection.
-			document.removeEventListener( 'selectionchange', this.onSelectionChange );
+		// Do not trigger a change if characters are being composed. Browsers
+		// will usually emit a final `input` event when the characters are
+		// composed.
```

Comment:
Thanks for all the comments. Maybe expand this one to reflect the current change, e.g.

> … As of December 2019, Safari doesn't honour `nativeEvent.isComposing`.

Incidentally, that could make it easier — later on, in a year or ten — to know what workarounds can be eliminated in our code base.

Reply:
Yes, I should add something here. I'm also wondering if I should still use `nativeEvent.isComposing` and fall back to `this.isComposing`. Normally `compositionstop` _should_ fire, but who knows, if in some rare cases it doesn't, the value would be stuck at `true` until composing is started again. `nativeEvent.isComposing` is probably more reliable. On the other hand, always using `this.isComposing` will more quickly surface any bugs.

Reply:
I don't oppose to listening to those two sources, but is there anything making you doubt of `compositionstop`'s reliability? Since it's a proper `SyntheticEvent`, I would expect adequate browser support.

Reply:
Yes, browser support is good, but I wouldn't be surprised if it's implemented slightly differently in browsers, just like the calling order of all these event handlers, or not called at all in some situations. Let's try this and keep an eye on reports.